file_optical.c

```diff
@@ -1670,13 +1670,68 @@ static int fbo_handle_cmd(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return 0;
 }
 
+static int fbo_update_cfgstring(struct tcmu_device *dev,
+				struct tcmulib_cfg_info *cfg)
+{
+	char *oldptr;
+	int ret;
+
+	if (!tcmu_check_state(dev, TCMU_DEV_STATE_CAPACITY_CHANGE_FOR_CONFIG_CHANGE)) {
```

Comment:
Yes, it is to enforce that if you swap the media and if it is different then the size is absorbed first. At first I didn't change the size and on media changes where the sizes change it causes a reset and the client isn't happy on either Linux or AIX.